src/components/file-indicator.js

```diff
@@ -0,0 +1,148 @@
+        fill: '#2755C4', // Primary-600
+      },
+      '&:active': {
+        fill: '#1B3EA4', // Primary-700
+      },
+    },
+    completePause: {
+      pointerEvents: 'none',
+      fill: '#366FE4', // Primary-500
+    },
+    disabledPause: {
+      pointerEvents: 'none',
+      fill: '#C8C8C8', // Secondary-400
+    },
+  })
+
+  return (
+    <div>
+      {
+        fileName && fileSize && <p className={`${style.fileName} text-secondary-700`}>{fileName} <span className={`${style.fileSize}`}>({fileSize})</span></p>
+      }
+      <div className={style.flexFile}>
+        <div ref={ref} className={fileIndicatorClasses.root} {...rest}>
+          {animate && <LinearAnimation
+            width={progress}
+            direction={direction}
+            duration={duration}
+            className={fileIndicatorClasses.content}
+          />}
+          {!animate && (
+            <div className={clsx(`${fileIndicatorClasses.content} ${style.fileIndicatorContent}`, {
+              'bg-secondary-500': error,
+            })} />
+          )}
+        </div>
+        <svg viewBox="0 0 16 15" fill="none" onClick={pause} 
+          className={clsx(`${style.pause}`, {
+            [style.pauseEnabled]: !error && progress !== 100,
+            [style.disabledPause]: error,
+            [style.completePause]: progress === 100,
+          })} xmlns="http://www.w3.org/2000/svg">
+          <path d="M8 0C6.51664 0 5.0666 0.439867 3.83323 1.26398C2.59986 2.08809 1.63856 3.25943 1.07091 4.62987C0.50325 6.00032 0.354725 7.50832 0.644114 8.96318C0.933503 10.418 1.64781 11.7544 2.6967 12.8033C3.7456 13.8522 5.08197 14.5665 6.53683 14.8559C7.99168 15.1453 9.49968 14.9968 10.8701 14.4291C12.2406 13.8614 13.4119 12.9001 14.236 11.6668C15.0601 10.4334 15.5 8.98336 15.5 7.5C15.5 5.51088 14.7098 3.60322 13.3033 2.1967C11.8968 0.790176 9.98913 0 8 0V0ZM11.125 10C11.125 10.1658 11.0592 10.3247 10.9419 10.4419C10.8247 10.5592 10.6658 10.625 10.5 10.625H5.5C5.33424 10.625 5.17527 10.5592 5.05806 10.4419C4.94085 10.3247 4.875 10.1658 4.875 10V5C4.875 4.83424 4.94085 4.67527 5.05806 4.55806C5.17527 4.44085 5.33424 4.375 5.5 4.375H10.5C10.6658 4.375 10.8247 4.44085 10.9419 4.55806C11.0592 4.67527 11.125 4.83424 11.125 5V10Z" 
+            className={clsx(`${style.pause}`, {
+              [style.pauseEnabled]: !error && progress !== 100,
+              [style.disabledPause]: error,
+              [style.completePause]: progress === 100,
+            })}/>
+        </svg>
+      </div>
+      {Text(style, error, progress, time)}
+    </div>
+  )
+})
+
+FileIndicator.propTypes = {
+  classes: PropTypes.object,
+  animate: PropTypes.bool,
+  direction: PropTypes.string,
+  duration: PropTypes.number,
+  fileName: PropTypes.string,
+  fileSize: PropTypes.string,
+  progress: PropTypes.number,
+  time: PropTypes.number,
+  pause: PropTypes.func,
+  error: PropTypes.bool,
+}
+
+FileIndicator.defaultProps = {
+  classes: { root: '', content: '' },
```

Comment:
we need a class selector for each of the container (at least for the main ones) so user can customize its styling. Please take a look on other component for example